u8timeseries/timeseries.py

```diff
@@ -804,6 +810,17 @@ def _combine_from_pd_ops(self, other: 'TimeSeries',
         conf_hi = self._combine_or_none(self._confidence_hi, other.conf_hi_pd_series(), combine_fn)
         return TimeSeries(series, conf_lo, conf_hi)
 
+    @staticmethod
+    def _fill_missing_dates(series: pd.Series) -> pd.Series:
+        date_axis = series.index
+        size = 3
+        available_frequencies = [date_axis[x:x + size].inferred_freq for x in range(len(date_axis) - size + 1)]
+        available_frequencies = set(filter(None.__ne__, available_frequencies))
+
+        raise_if_not(len(available_frequencies) == 1, 'Could not infer explicit frequency.', logger)
```

Comment:
Should this be `len(available_frequencies) >= 1` ?

Reply:
If there is more than one for instance 2D and 3D then we cannot decide, I would suggest to fail in such case